### DISCUSSION TITLE
Zustand breaks @react-google-maps/api for some reason

### DISCUSSION THREAD

**Discussion by davidarny** [pmndrs/zustand#1900] [Bug report] [answered]
Hey!

I've spend hours trying to find out why this happens. In short, for some reason google maps break when adding `zustand` dependency. Here is a reproduction of that issue -> https://codesandbox.io/s/zustand-google-maps-ykgkx2

You can try to comment out `L:33` and `_points` variable in `L:54` and cluster will be rendered. This is not a case when using zustand stores. There may be some side effect in `zustand`'s code (or vice versa), but I haven't found anything.

**Comment by dai-shi** [comment]
Can anyone help?

**Reply by dbritto-dev** [reply]
@DavidArutiunian I can help but I need a temporal access token (max 2 days)

**Reply by dbritto-dev** [reply, marked as answer]
@DavidArutiunian  here you go: https://codesandbox.io/s/zustand-google-maps-forked-x4w77d?file=/src/App.js

**Reply by davidarny** [reply]
@dbritto-dev so do you understand why doesn't it work when everything is flattened?

**Reply by dbritto-dev** [reply]
@DavidArutiunian yeah, try not to use anything that could trigger a new render on the map that could lead you to some bugs

**Reply by dbritto-dev** [reply]
What I did is just remove that from top of app component to avoid that and use container pattern to make it simple to use
